tests/test_migrate_13_0_to_14_0.py

```diff
@@ -0,0 +1,309 @@
+
+        # 2. Published record with draft
+        draft = records_service.create(system_identity, thesis_record_data)
+        record_with_draft = records_service.publish(system_identity, draft.id)
+
+        # Create draft for the published record
+        draft_with_record = records_service.edit(system_identity, record_with_draft.id)
+        draft_with_record.data["metadata"]["title"] = "Updated Title in Draft"
+        records_service.update_draft(
+            system_identity, draft_with_record.id, draft_with_record.data
+        )
+
+        # 3. Records with multiple versions
+        old_version_draft = records_service.create(system_identity, thesis_record_data)
+        old_version_record = records_service.publish(
+            system_identity, old_version_draft.id
+        )
+
+        new_version_draft = records_service.new_version(
+            system_identity, old_version_record.id
+        )
+        new_version_draft.data["metadata"][
+            "title"
+        ] = "Updated Title in New Version Record"
+        new_version_draft.data["metadata"]["publication_date"] = "2024-02-01"
+        new_version_draft = records_service.update_draft(
+            system_identity, new_version_draft.id, new_version_draft.data
+        )
+        new_version_record = records_service.publish(
+            system_identity, new_version_draft.id
+        )
+
+        old_version_draft = records_service.edit(system_identity, new_version_record.id)
+        old_version_draft.data["metadata"][
+            "title"
+        ] = "Updated Title in Old Version Draft"
+        old_version_draft = records_service.update_draft(
+            system_identity, old_version_draft.id, old_version_draft.data
+        )
+
+        # 4. Draft-only records
+        draft_only = records_service.create(system_identity, thesis_draft_data)
+
+        # 5. Non-thesis record (should not be affected)
+        thesis_record_data["metadata"]["resource_type"]["id"] = "publication-article"
+        non_thesis_draft = records_service.create(system_identity, thesis_record_data)
+        non_thesis_record = records_service.publish(
+            system_identity, non_thesis_draft.id
+        )
+
+        non_thesis_draft = records_service.create(system_identity, thesis_record_data)
+        non_thesis_draft.data["metadata"]["title"] = "Updated Title in Non-Thesis Draft"
+        non_thesis_draft = records_service.update_draft(
+            system_identity, non_thesis_draft.id, non_thesis_draft.data
+        )
+
+        # Run the complete migration
+        execute_upgrade()
+
+        # Verify all thesis records were migrated
+        thesis_records = records_service.search(
+            system_identity,
+            params={"allversions": True},
+            extra_filter=dsl.Q(
+                "term", **{"metadata.resource_type.id": "publication-thesis"}
+            ),
+        )
+        assert thesis_records.total == 0
```

Comment:
When a draft is published as record, the draft is soft deleted right? But it comes up in the search here. Is that expected? The test is failing here.